pkg/backend/metrics.go

```diff
@@ -0,0 +1,203 @@
+	dimensions["resourcegroup"] = doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+	dimensions["name"] = doc.OpenShiftCluster.Name
+	dimensions["ocpversion"] = doc.OpenShiftCluster.Properties.ClusterProfile.Version
+	dimensions["workercount"] = strconv.FormatInt(int64(len(doc.OpenShiftCluster.Properties.WorkerProfiles)), 10)
+	dimensions["rpversion"] = doc.OpenShiftCluster.Properties.ProvisionedBy
+
+	for flag, feature := range doc.OpenShiftCluster.Properties.OperatorFlags {
+		dimensions[fmt.Sprintf("operator-%s", flag)] = feature
+	}
+
+	if doc.OpenShiftCluster.Tags != nil {
+		dimensions["tags"] = enabled
+	} else {
+		dimensions["tags"] = disabled
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherNodeMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.OpenShiftCluster.Properties.MasterProfile.DiskEncryptionSetID != "" {
+		dimensions["masterprofile.diskencryptionsetid"] = enabled
+	} else {
+		dimensions["masterprofile.diskencryptionsetid"] = disabled
+	}
+
+	if doc.OpenShiftCluster.Properties.MasterProfile.EncryptionAtHost == api.EncryptionAtHostEnabled {
+		dimensions["masterprofile.encryptionathost"] = string(api.EncryptionAtHostEnabled)
+	} else if doc.OpenShiftCluster.Properties.MasterProfile.EncryptionAtHost == api.EncryptionAtHostDisabled {
+		dimensions["masterprofile.encryptionathost"] = string(api.EncryptionAtHostDisabled)
+	} else {
+		dimensions["masterprofile.encryptionathost"] = unknown
+	}
+
+	for _, p := range doc.OpenShiftCluster.Properties.WorkerProfiles {
+		if p.VMSize != api.VMSizeStandardD4sV3 {
+			dimensions["customvmsize"] = string(p.VMSize)
+		}
+
+		if p.DiskSizeGB > 128 {
+			dimensions["customisksizegb"] = strconv.FormatInt(int64(p.DiskSizeGB), 10)
+		}
+
+		if p.EncryptionAtHost == api.EncryptionAtHostEnabled {
+			dimensions["workprofiles.encryptionathost"] = fmt.Sprintf("%s.%s", string(api.EncryptionAtHostEnabled), p.Name)
+		} else if p.EncryptionAtHost == api.EncryptionAtHostDisabled {
+			dimensions["workprofiles.encryptionathost"] = fmt.Sprintf("%s.%s", string(api.EncryptionAtHostDisabled), p.Name)
+		} else {
+			dimensions["workprofiles.encryptionathost"] = unknown
+		}
+	}
+
+	if doc.OpenShiftCluster.Properties.ClusterProfile.FipsValidatedModules == api.FipsValidatedModulesEnabled {
+		dimensions["fips"] = string(api.FipsValidatedModulesEnabled)
+	} else if doc.OpenShiftCluster.Properties.ClusterProfile.FipsValidatedModules == api.FipsValidatedModulesDisabled {
+		dimensions["fips"] = string(api.FipsValidatedModulesDisabled)
+	} else {
+		dimensions["fips"] = unknown
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherAuthMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil {
+		dimensions["workloadidentity"] = enabled
+	} else if doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
+		dimensions["serviceprincipal"] = disabled
```

Comment:
this should be enabled, right?

Reply:
I'm also wondering if we should instead have a single dimension here named something like `clusteridentity` that is set to either `serviceprincipal` or `managed`?

Reply:
That makes more sense, updated with your suggestion.